File: 251002/Zhukovskaya/Lab1-5/publisher/src/main/java/com/rita/publisher/model/Tweet.java
package com.rita.publisher.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name="tbl_tweet")
public class Tweet {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name="writer_id", nullable = false)
    private Writer writer;
    @Column(nullable = false,unique = true)
    private String title;
    private String content;
    private LocalDateTime createdTime;
    private LocalDateTime modifiedTime;
    @ManyToMany(mappedBy = "tweets")
    private Set<Sticker> stickers=new HashSet<>();
//    @OneToMany(fetch = FetchType.LAZY, mappedBy = "tweet",
//            cascade = CascadeType.ALL)
//    private List<Message> messages=new ArrayList<Message>();
}

File: 251003/Kukhatskavolets/microblog/publisher/src/main/kotlin/by/kukhatskavolets/publisher/entities/Tweet.kt
package by.kukhatskavolets.publisher.entities

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "tbl_tweet", schema = "public")
data class Tweet(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", nullable = false)
    val user: User = User(),

    @Column(nullable = false, unique = true, length = 64)
    val title: String = "",

    @Column(nullable = false, length = 2048)
    val content: String = "",

    @Column(nullable = false)
    val created: LocalDateTime = LocalDateTime.now(),

    @Column(nullable = false)
    var modified: LocalDateTime = LocalDateTime.now(),

    @OneToMany(mappedBy = "tweet", cascade = [CascadeType.ALL], orphanRemoval = true)
    val comments: MutableList<Comment> = mutableListOf(),

    @ManyToMany(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinTable(
        name = "tbl_tweet_mark",
        schema = "public",
        joinColumns = [JoinColumn(name = "tweetId")],
        inverseJoinColumns = [JoinColumn(name = "markId")]
    )
    val marks: MutableSet<Mark> = mutableSetOf()
)
